Update Flutter go_router documentation (#37039)
DataDog/documentation: layouts/shortcodes/mdoc/en/sdk/integrated_libraries/flutter.mdoc.md

);
```

> If you are using a version of `go_router` prior to version 17.x and you are using ShellRoutes, you need to supply a separate observer to each `ShellRoute`, as shown below. Post version 17 of `go_router`, this workaround is no longer needed. See [this bug][3] for more information.

Reviewer: ```suggestion
If you are using `go_router` earlier than version 17 and are using ShellRoutes, you need to supply a separate observer to each `ShellRoute`, as shown below. In version 17 and later, this workaround is no longer needed. See [this bug][3] for more information.
```
